src/plugins/data/common/search/search_source/search_source.ts

```diff
@@ -116,6 +116,11 @@ export class SearchSource {
   constructor(fields: SearchSourceFields = {}, dependencies: SearchSourceDependencies) {
     this.fields = fields;
     this.dependencies = dependencies;
+
+    if (fields.parent) {
+      this.setParent(new SearchSource(fields.parent, dependencies));
+      delete this.fields.parent;
+    }
```

Comment:
Maybe better to write this in a way that does not mutate `fields` param.

```suggestion
  constructor(fields: SearchSourceFields = {}, dependencies: SearchSourceDependencies) {
    const { parent, ...currentFields } = fields;
    this.fields = currentFields;
    this.dependencies = dependencies;

    if (parent)
      this.setParent(new SearchSource(parent, dependencies));
  }
```